packages/insights/src/db/query.ts

```diff
@@ -167,3 +167,40 @@ export async function updateEdge(
     await db.insert(edgeTable).values(edgeRow).run();
   }
 }
+
+export async function updateRoutes(
```

Comment:
look at `updateEdge` that function is much more efficient as it only needs a single SQL to achieve the goal rather than your function requiring two.

Reply:
🤔 I looked at the updateEdge implementation but in this case I need to update interactions based on what is in the database.

Reply:
FIxed [e739e8f](https://github.com/BuilderIO/qwik/pull/4971/commits/e739e8f10d15eb6a0a8763d5025210182f07b4a5)